tough/src/transport.rs

```diff
@@ -140,14 +140,26 @@ pub struct FilesystemTransport;
 
 impl Transport for FilesystemTransport {
     fn fetch(&self, url: Url) -> Result<Box<dyn Read + Send>, TransportError> {
+        // If the scheme isn't "file://", reject
         if url.scheme() != "file" {
             return Err(TransportError::new(
                 TransportErrorKind::UnsupportedUrlScheme,
                 url,
             ));
         }
 
-        let f = std::fs::File::open(url.path()).map_err(|e| {
+        // Convert the file URL into a file path
+        let file_path = &url.to_file_path().map_err(|_e| {
+            TransportError::new(TransportErrorKind::Other, &url)
+            // We return no source because :
+            // - The source must implement Error, but url.to_file_path() only returns a () as a error return value.
+            // - We can't write an oneoff error like anyhow!("Failed to convert the URL into a file path") because anyhow isn't listed as a dep, but if added ;
+            //   that would be replaced with :
+            // TransportError::new_with_cause((TransportErrorKind::Other, url, anyhow::anyhow!("Conversion from URL to disk path failed"))
+        })?;
```

Comment:
OK, thank you for explaining that in the comment. The problem is that there would be no good message. Rust implements Error for String, so I think this would work, and would provide an error message to the user.

```suggestion
        // Convert the file URL into a file path
        let file_path = &url.to_file_path().map_err(|_| {
            TransportError::new_with_source(TransportErrorKind::Other, &url, "unable to get filepath from URL".to_string())
        })?;
```

Reply:
Ok, committing that, I didn't think about this one, idk why.